### PR TITLE
feat(bbmap): Initialize pileup module

### PR DESCRIPTION
**Pull request by Emiller88**
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: tests/modules/bbmap/pileup/test.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+- name: "bbmap pileup"
+  command: nextflow run ./tests/modules/bbmap/pileup -entry test_bbmap_pileup -c ./tests/config/nextflow.config -c ./tests/modules/bbmap/pileup/nextflow.config
+  tags:
+    - "bbmap"
+    - "bbmap/pileup"
+  files:
+    - path: "output/bbmap/test.bam"
```
Comment (rpetit3) [inline tests/modules/bbmap/pileup/test.yml:7]: Will need to be updated to test for stats and hist outputs

**Review by rpetit3 — CHANGES_REQUESTED** [review]
Everything looked good! Just need add the outputs to the test.yml

File: tests/modules/bbmap/pileup/test.yml (hunk 1 above)
Comment (Emiller88) [reply tests/modules/bbmap/pileup/test.yml:7]: Good catch, got derailed when the CI was down this morning!

**Review by rpetit3 — APPROVED** [review]

**Comment by rpetit3** [comment]
Lgtm!

**Pull request merged**